runtime/runtime.go

```diff
@@ -75,25 +77,48 @@ func (c *Context) InitializeCodesAndPrograms() {
 	}
 }
 
+// Runnable can be run by a runner (including transactions, system operations, or read only scripts)
+// includes all the read-only data provided for the execution
```

Comment:
The data provided is readonly, but since this could be a transaction there could be changes to the state right?

Reply:
yeah, probably my note is misleading, this doesn't mean the operation is read-only, it means the values provided by runnable won't change in the middle of execution (such as source, arguments, authorizers)